### DISCUSSION TITLE
Deprecated SubProgressMonitor is a mess

### DISCUSSION THREAD

**Discussion by jukzi** [eclipse-equinox/equinox#392] [Ideas]
I gave it a try to refactor jdt to SubMonitor but there seems to be no easy adaptation of the constructors where 
org.eclipse.core.runtime.SubProgressMonitor.SUPPRESS_SUBTASK_LABEL / PREPEND_MAIN_LABEL_TO_SUBTASK
is used.
I suggest to de-deprecate SubMonitor , as nobody seems to truly convert the calls and the plenty deprecation warnings do not help.

**Comment by laeubi** [comment]
Maybe we can add the missing thing? `SubProgressMonitor.SUPPRESS_SUBTASK_LABEL` by the way should be replaced by `SubMonitor.split(..., SUPPRESS_SUBTASK | <maybe more flags>`)

> I suggest to de-deprecate SubMonitor , as nobody seems to truly convert the calls and the plenty deprecation warnings do not help.

Just because JDT do not do this does not mean nobody is doing that. e.g. in rest of platform I see only a few references left and deprecation also serves the purpose of preventing new code from using it.

**Reply by jukzi** [reply]
There are 593 deprecation warnings about SubProgressMonitor without any obvious migration path in the IDE workspace. Thats more then a few.

**Reply by vogella** [reply]
IIRC I did some migration of this in the past together with @kthoms, but unfortunately not a lot did join the effort. Deprecation is still good to avoid new usage of this API and to help new "cleaners" to identify work.

**Comment by jukzi** [comment]
Mind if i offer an ProgressMonitor.split(int, Enum ... flags) API which does  NOT require to manually construct a SubMonitor - similar to ProgressMonitor.slice(int) for non-concurrent use-case?

**Reply by laeubi** [reply]
Sounds reasonable but might be hard to default implement correctly, be aware that `IProgressMonitor` is API implemented on different places!

> Clients may implement this interface.
